index.js

```diff
@@ -0,0 +1,185 @@
+'use strict'
```

Comment:
some part of this code is copied from Node.js. We need to retain their license here at the top.